src/isolate/server/server.py

```diff
@@ -39,35 +39,51 @@ def Run(
         context: ServicerContext,
     ) -> Iterator[definitions.PartialRunResult]:
         messages: Queue[definitions.PartialRunResult] = Queue()
-        try:
-            environment = from_grpc(request.environment)
-        except ValueError:
-            return self.abort_with_msg(
-                f"Unknown environment kind: {request.environment.kind}.",
-                context,
-            )
-        except TypeError as exc:
-            return self.abort_with_msg(
-                f"Invalid environment parameter: {str(exc)}.",
-                context,
-            )
+        environments = []
+        for env in request.environments:
+            try:
+                environments.append(from_grpc(env))
+            except ValueError:
+                return self.abort_with_msg(
+                    f"Unknown environment kind",
+                    context,
+                )
+            except TypeError as exc:
+                return self.abort_with_msg(
+                    f"Invalid environment parameter: {str(exc)}.",
+                    context,
+                )
+
+        assert len(environments), "No environment definition found."
```

Comment:
I'd recommend making this a proper error, like the validation errors above.
```suggestion
        if not environments:
            return self.abort_with_msg(
                f"At least one environment must be specified for a run!",
                context,
            )